go/vt/vtgate/planbuilder/querytree_transformers.go

```diff
@@ -464,6 +466,11 @@ func transformRoutePlan(ctx *planningContext, n *routeTree) (*route, error) {
 }
 
 func transformJoinPlan(ctx *planningContext, n *joinTree) (logicalPlan, error) {
+	canHashJoin, lhsInfo, rhsInfo, err := canHashJoin(ctx, n)
```

Comment:
we should use the cost model here to choose between an Apply join and a Hash join